api/v1/controllers/RSVPController.js

```diff
@@ -0,0 +1,146 @@
+var bodyParser = require('body-parser');
+
+var services = require('../services');
+var middleware = require('../middleware');
+var requests = require('../requests');
+var roles = require('../utils/roles');
+var mail = require('../utils/mail');
+var User = require('../models/User');
+
+var router = require('express').Router();
+
+function _isAuthenticated (req) {
+    return req.auth && (req.user !== undefined);
+}
+
+function _removeFromList(rsvpCurrent, rsvpNew) {
+    if(rsvpCurrent.get('attendeeAttendance') !== 'NO' && rsvpNew.attendeeAttendance === 'NO')
+        return true;
+    return false;
+}
+
+function _addToList(rsvpCurrent, rsvpNew) {
+    if(rsvpCurrent.get('attendeeAttendance') === 'NO' && rsvpNew.attendeeAttendance !== 'NO')
+        return true;
+    return false;
+}
+
+function createRSVP(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .createRSVP(attendee, req.user, req.body)
+                .then(function (rsvp) {
+                    if(rsvp.get('attendeeAttendance') !== 'NO')
+                        services.MailService.addToList(req.user, mail.lists.attendees);
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        });
+}
+
+function fetchRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function fetchRSVPByAttendeeId(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function updateRSVPByUser(req, res, next) {
+    services.RegistrationService
```

Comment:
This looks pretty identical to the create RSVP handler. Is there a way you can refactor this so that you have a helper function used by both the create and update RSVP handlers?